File: run-planner-backend/src/main/java/com/runplanner/workout/Workout.java
package com.runplanner.workout;

import com.runplanner.user.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "workouts")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Workout {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String source;

    private String sourceId;

    @Column(nullable = false)
    private Instant startedAt;

    @Column(nullable = false)
    private Double distanceMeters;

    @Column(nullable = false)
    private Integer durationSeconds;

    private Integer avgHr;

    private Integer maxHr;

    private Double elevationGain;
}

File: run_planner_app/lib/models/workout.dart
class WorkoutResponse {
  final String id;
  final DateTime startedAt;
  final double distanceMeters;
  final int durationSeconds;
  final int? avgHr;
  final int? maxHr;
  final double? elevationGain;
  final String source;
  final String sourceId;

  const WorkoutResponse({
    required this.id,
    required this.startedAt,
    required this.distanceMeters,
    required this.durationSeconds,
    this.avgHr,
    this.maxHr,
    this.elevationGain,
    required this.source,
    required this.sourceId,
  });

  double get paceMinPerKm {
    if (distanceMeters <= 0) return 0;
    return (durationSeconds / 60) / (distanceMeters / 1000);
  }

  factory WorkoutResponse.fromJson(Map<String, dynamic> json) {
    return WorkoutResponse(
      id: json['id'] as String,
      startedAt: DateTime.parse(json['startedAt'] as String),
      distanceMeters: (json['distanceMeters'] as num).toDouble(),
      durationSeconds: json['durationSeconds'] as int,
      avgHr: json['avgHr'] as int?,
      maxHr: json['maxHr'] as int?,
      elevationGain: (json['elevationGain'] as num?)?.toDouble(),
      source: json['source'] as String,
      sourceId: json['sourceId'] as String,
    );
  }
}
